schematic/configuration/dataclasses.py

```diff
@@ -0,0 +1,189 @@
+        """
+        if len(value) == 0:
+            raise ValueError(f"{value} is an empty string")
+        return value
+
+
+@dataclass(config=pydantic_config)
+class ManifestConfig:
+    """
+    title:
+    data_type:
+    """
+
+    title: str = "example"
+    data_type: list[str] = field(default_factory=lambda: ["Biospecimen", "Patient"])
+
+    @validator("title")
+    @classmethod
+    def validate_string_is_not_empty(cls, value: str) -> str:
+        """Check if string  is not empty(has at least one char)
+
+        Args:
+            value (str): A string
+
+        Raises:
+            ValueError: If the value is zero characters long
+
+        Returns:
+            (str): The input value
+        """
+        if len(value) == 0:
+            raise ValueError(f"{value} is an empty string")
+        return value
+
+
+@dataclass(config=pydantic_config)
+class ModelConfig:
+    """
+    location: location of the schema jsonld, either a path, ro url
+    file_type: one of ["local"]
+    """
+
+    location: str = "tests/data/example.model.jsonld"
+    file_type: str = "local"
+
+    @validator("location", "file_type")
+    @classmethod
+    def validate_string_is_not_empty(cls, value: str) -> str:
+        """Check if string  is not empty(has at least one char)
+
+        Args:
+            value (str): A string
+
+        Raises:
+            ValueError: If the value is zero characters long
+
+        Returns:
+            (str): The input value
+        """
+        if len(value) == 0:
+            raise ValueError(f"{value} is an empty string")
+        return value
+
+
+@dataclass(config=pydantic_config)
+class GoogleSheetsConfig:
+    """
+    master_template_id:
+    strict_validation:
+    service_acct_creds_synapse_id:
+    service_acct_creds_basename:
+    """
+
+    service_acct_creds_synapse_id: str = "syn25171627"
+    service_acct_creds_basename: str = "schematic_service_account_creds.json"
+    master_template_id: Optional[str] = None
+    strict_validation: bool = True
+
+    @validator("service_acct_creds_basename")
+    @classmethod
+    def validate_string_is_not_empty(cls, value: str) -> str:
+        """Check if string is not empty(has at least one char)
+
+        Args:
+            value (str): A string
+
+        Raises:
+            ValueError: If the value is zero characters long
+
+        Returns:
+            (str): The input value
+        """
+        if len(value) == 0:
+            raise ValueError(f"{value} is an empty string")
+        return value
+
+    @validator("master_template_id")
+    @classmethod
+    def validate_optional_string_is_not_empty(cls, value: str) -> str:
+        """Check if string is not empty(has at least one char)
+
+        Args:
+            value (Optional[str]): A string
+
+        Raises:
+            ValueError: If the value is zero characters long
+
+        Returns:
+            (str): The input value
+        """
+        if value is not None and len(value) == 0:
```

Comment:
This can be simplify to: 
```
if not value
```

Reply:
Yes I think you are correct

Reply:
hmm but your original statement is: 
```
if value is not None
```

Reply:
Could you elaborate on your point?

Reply:
It seems like you have made the change. Let me mark this as resolved